generator/src/test/java/com/scottlogic/deg/generator/cucumber/GreaterThan.feature

```diff
@@ -6,6 +6,396 @@ Feature: User can specify that a numeric value is higher than, but not equal to,
+      | foo         |
+      | -2147482624 |
+      | -2147482623 |
+      | -2147482622 |
+      | -2147482621 |
+      | -2147482620 |
+
+Scenario: not greaterThan run against a non contradicting not lessThan should be successful (not greaterThan 5 AND not lessThan 4)
+    Given foo is anything but greater than 5
+    And foo is anything but less than 4
+    Then the following data should be generated:
+      | foo |
+      | 4   |
+      | 5   |
+
+Scenario: greaterThan run against a contradicting lessThan should only only generate string, temporal and null (greaterThan 2 AND lessThan 2)
+    Given foo is greater than 2
+    And foo is less than 2
+    Then no data is created
+
+Scenario: not greaterThan run against a contradicting not lessThan should only only generate string, temporal and null (not greaterThan 2 AND not lessThan 2)
+    Given foo is anything but greater than 2
+    And foo is anything but less than 2
+    Then the following data should be generated:
+      | foo |
+      | 2   |
+
+#lessThanOrQualTo
+Scenario: greaterThan run against a non contradicting lessThanOrEqualTo should be successful (greaterThan 2 AND lessThanOrEqualTo 5)
+    Given foo is greater than 2
+    And foo is less than or equal to 5
+    Then the following data should be generated:
+      | foo |
+      | 3   |
+      | 4   |
+      | 5   |
+
+Scenario: greaterThan run against a non contradicting not lessThanOrEqualTo should be successful (greaterThan 1 AND not lessThanOrEqualTo 2)
+    Given foo is greater than 1
+    And foo is anything but less than or equal to 2
+    And the generator can generate at most 5 rows
+    Then the following data should be generated:
+      | foo |
+      | 3   |
+      | 4   |
+      | 5   |
+      | 6   |
+      | 7   |
+
+@ignore #589: produces big-integer numbers rather than integer values, not sure why
+Scenario: not greaterThan run against a non contradicting lessThanOrEqualTo should be successful (not greaterThan 4 AND lessThanOrEqualTo 2)
+    Given foo is anything but greater than 4
+    And foo is less than or equal to 2
+    And the generator can generate at most 5 rows
+    Then the following data should be generated:
+      | foo         |
+      | -2147482624 |
+      | -2147482623 |
+      | -2147482622 |
+      | -2147482621 |
+      | -2147482620 |
+
+Scenario: not greaterThan run against a non contradicting not lessThanOrEqualTo should be successful (not greaterThan 5 AND not lessThanOrEqualTo 4)
+    Given foo is anything but greater than 5
+    And foo is anything but less than or equal to 4
+    And the generator can generate at most 5 rows
+    Then the following data should be generated:
+      | foo |
+      | 5   |
+
+Scenario: greaterThan run against a contradicting lessThanOrEqualTo should only only generate string, temporal and null (greaterThan 2 AND lessThanOrEqualTo 1)
+    Given foo is greater than 2
+    And foo is less than or equal to 1
+    Then no data is created
```

Comment:
Remove null from the background and test that null is created here to show that the logic works as expected rather than throwing an error.